core/src/main/java/org/apache/iceberg/avro/BaseWriteBuilder.java

```diff
@@ -64,6 +64,11 @@ public ValueWriter<?> map(Schema map, ValueWriter<?> valueWriter) {
     return ValueWriters.map(ValueWriters.strings(), valueWriter);
   }
 
+  @Override
+  public ValueWriter<?> variant(Schema variant, List<ValueWriter<?>> fields) {
+    return createRecordWriter(fields);
+  }
+
```

Comment:
To test out the write, It needs the changes in https://github.com/apache/iceberg/pull/11934/commits/125fa6a56d6b6d741d1bc9c5cf9fe94825ffa44a so I haven't added the coverage yet.